### PR TITLE
Fix/flip group

### PR DESCRIPTION
**Pull request by judicaelandria**

### CODE REVIEW

**Comment by judicaelandria** [comment]
closes https://github.com/tldraw/tldraw/issues/878

**Comment by steveruizok** [comment]
So @ds300 and I went through this UX for the next version, and settled on:
- if a group is the only selected shape, then flip should effect its children
- otherwise, flip should not effect the children

**Comment by steveruizok** [comment]
Also, undo seems broken:

- create a group
- create another shape
- select both the group and the shape
- flip
- undo

File: packages/tldraw/src/state/TLDR.ts [hunk 1]
```diff
@@ -394,26 +395,46 @@ export class TLDR {
 
     const shapes = data.document.pages?.[data.appState.currentPageId].shapes
     const groupShape = shapes?.[ids[0]]
+    const isSingle = ids.length === 1
+
+    const mirror = (data: TDSnapshot, shapeId: string, pageId: string, index: number) => {
```
Comment (ds300) [inline packages/tldraw/src/state/TLDR.ts:400]: ah I see now why I was so confused yesterday, it looks like this `mutateShapes` function _is_ supposed to be generic, and it shouldn't know anything about flipping shapes. All the flipping logic you've added here should really live in the `flipShapes.ts` file. That should simplify the logic quite a lot too. There might still need to be one or two small changes in this file to allow it to work properly (e.g. for treating children differently when the group is not the only selected shape) but the diff in this file should be minimal. Let me know if you want to pair on this, i'm happy to help :)

**Pull request merged**